portable/ThirdParty/GCC/Posix/port.c

```diff
@@ -51,9 +51,6 @@
 *----------------------------------------------------------*/
 #include "portmacro.h"
 
-#ifdef __linux__
-    #define __USE_GNU
```

Comment:
Did you try changing this to `#define _GNU_SOURCE`?

Reply:
I tried two compilation environments, but they both reported warnings
![image](https://github.com/FreeRTOS/FreeRTOS-Kernel/assets/65268883/8bef75a9-8a1e-4941-9d77-7d5f619af17a)

Reply:
Does it happen even if you remove this `#ifdef __linux__` check?

Reply:
yes
![image](https://github.com/FreeRTOS/FreeRTOS-Kernel/assets/65268883/999dad14-6f51-4086-80dc-ba48592a335a)
![image](https://github.com/FreeRTOS/FreeRTOS-Kernel/assets/65268883/c91c0bc8-2d8c-4c16-b741-e8c10c0c0bff)

Reply:
And which OS are you using? Is it Ubuntu?

Reply:
yes，ubuntu:22.04

Reply:
Another is a docker

Reply:
That is strange because I do not see this warning on Ubuntu 22.04. Can you please do the following:

1. Locate pthread.h:
    ```
    locate pthread.h 
    ```
2. Open it and find `pthread_setname_np` in it.

My pthread.h is located at `/usr/include/pthread.h` and I see the following declaration in it:
```
#ifdef __USE_GNU
/* Get thread name visible in the kernel and its interfaces.  */
extern int pthread_getname_np (pthread_t __target_thread, char *__buf,
                               size_t __buflen)
     __THROW __nonnull ((2));

/* Set thread name visible in the kernel and its interfaces.  */
extern int pthread_setname_np (pthread_t __target_thread, const char *__name)
     __THROW __nonnull ((2));
#endif
```

Reply:
My file ptthread. h is the same as yours. I tried a pure ubuntu22.04 Docker, but it still has this warning. The test log is as follows
[test_ubuntu22.04.txt](https://github.com/user-attachments/files/15688487/test_ubuntu22.04.txt)

Reply:
This does not make sense - Either `__USE_GNU` is not getting defined correctly or you are using a different pthread.h file. Can you dump the preprocessed port.c file?

Reply:
[port.zip](https://github.com/user-attachments/files/15730120/port.zip)
![image](https://github.com/FreeRTOS/FreeRTOS-Kernel/assets/65268883/243d13cc-e4d8-4d09-8715-d3a5b31bbcac)
This is my "port. c" and compiled. o files, as well as the "ptthread. h" and "features. h" in the system library

Reply:
Would you please use the slightly modified port.c: [port.zip](https://github.com/user-attachments/files/15756812/port.zip). Also, please dump the pre-processed port.c file ans share - you can use gcc `-E` option for that.

Reply:
[port.zip](https://github.com/user-attachments/files/15757234/port.zip)
this is use gcc -E

Reply:
Did you use the modified file I shared? I changed `_GNU_SOURCE` to `__USE_GNU` in port.c:
```
#define __USE_GNU
```

If you want to keep using `_GNU_SOURCE`, please move that to before including `portmacro.h`:

```c
#define _GNU_SOURCE

#include "portmacro.h"
#include <errno.h>
#include <pthread.h>
#include <limits.h>
#include <signal.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <sys/time.h>
#include <sys/times.h>
#include <time.h>
#include <unistd.h>
```

Reply:
i move _GNU_SOURCE that to before including portmacro.h, No warning is reported, and it can be compiled on gcc 6.3 and glibc 2.24

Reply:
Thank you for following up and confirming.